node/npm_dependencies.ts

```diff
@@ -14,6 +15,51 @@ import { Logger } from './logger.js'
 
 const TYPESCRIPT_EXTENSIONS = new Set(['.ts', '.cts', '.mts'])
 
+/**
+ * Returns the name of the `@types/` package used by a given specifier. Most of
+ * the times this is just the specifier itself, but scoped packages suffer a
+ * transformation (e.g. `@netlify/functions` -> `@types/netlify__functions`).
+ * https://github.com/DefinitelyTyped/DefinitelyTyped#what-about-scoped-packages
+ */
+const inferDefinitelyTypedPackage = (specifier: string) => {
+  if (!specifier.startsWith('@')) return `@types/${specifier}`
+  const [scope, pkg] = specifier.split('/')
+  return `@types/${scope.replace('@', '')}__${pkg}`
+}
+
+/**
+ * Starting from a `package.json` file, this tries detecting a typescript declaration file.
+ * It first looks at the `types` and `typings` fields in `package.json`.
+ * If it doesn't find them, it falls back to DefinitelyTyped packages (`@types/...`).
+ */
+const detectTypes = async (packageJsonPath: string): Promise<string | undefined> => {
```

Comment:
nit: I think the name of the method could be a bit more descriptive. "detect" doesn't tell me much about what this returns. I would consider calling this something like `getTypesPath`.

Reply:
renamed in https://github.com/netlify/edge-bundler/pull/505/commits/a0a27f0cc7a1beee318a17642450fefe09afcb66